src/CMakeLists.txt

```diff
@@ -51,28 +51,11 @@ if( WIN32 )
 	add_definitions( -D_WIN32 )
 
 
-	if( ( MSVC14 AND NOT CMAKE_GENERATOR_TOOLSET STREQUAL "v140_xp" ) OR	# For VS 2015.
-		( MSVC15 AND NOT CMAKE_GENERATOR_TOOLSET STREQUAL "v141_xp" ) )		# For VS 2017.
+	if( MSVC AND MSVC_VERSION GREATER 1699 )	# For VS 2015 and later.
```

Comment:
`MSVC_VERSION` is equal to 1700 [for VS2012](https://cmake.org/cmake/help/v3.16/variable/MSVC_VERSION.html). Moreover, we do not support VS2015 anymore.